src/platform/storage_winuser.c

```diff
@@ -348,5 +348,116 @@ CxPlatStorageClear(
     _In_ CXPLAT_STORAGE* Storage
     )
 {
-    return HRESULT_FROM_WIN32(RegDeleteTreeA(Storage->RegKey, NULL));
-}
+    QUIC_STATUS Status = QUIC_STATUS_SUCCESS;
+    DWORD Error = NO_ERROR;
+    DWORD AllocatedLength = 255;
+    PSTR ValueName = NULL;
+
+    ValueName = CXPLAT_ALLOC_PAGED(AllocatedLength, QUIC_POOL_PLATFORM_TMP_ALLOC);
+    if (ValueName == NULL) {
+        Status = QUIC_STATUS_OUT_OF_MEMORY;
+        QuicTraceEvent(
+            AllocFailure,
+            "Allocation of '%s' failed. (%llu bytes)",
+            "RegEnumValueA ValueName",
+            AllocatedLength);
+        goto Exit;
+    }
+
+    //
+    // Iterate through all values and delete them
```

Comment:
I expect the old RegDeleteTreeA would recurse, too. Is that true, and would we ever rely on the recursion?

Reply:
Thanks for the context. Let's put that design decision (and now API contract) into the comments for this function.

(Bonus points if you have a link to your original comment handy)